extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/theta/SketchBufferAggregator.java

```diff
@@ -30,17 +30,16 @@
 
 import java.nio.ByteBuffer;
 import java.util.HashMap;
+import java.util.IdentityHashMap;
 import java.util.Map;
 
 public class SketchBufferAggregator implements BufferAggregator
 {
   private final ObjectColumnSelector selector;
   private final int size;
   private final int maxIntermediateSize;
-
-  private NativeMemory nm;
-
   private final Map<Integer, Union> unions = new HashMap<>(); //position in BB -> Union Object
+  private final IdentityHashMap<ByteBuffer, NativeMemory> nmCache = new IdentityHashMap<>();
```

Comment:
we need to keep this map b/c `BufferAggregator` does not guarantee ByteBuffer passed in any of the methods `aggregate`, `relocate` is going to be the same. There might be some implementations which can send different ByteBuffer and to support that we need to cache those ByteBuffers.
Given that, I will use to the original interface `relocate(int oldPosition, int new Position, ByteBuffer newBuffer)` b/c diff relocate calls can get diff buffer's.